### PR TITLE
add cost_aware_partition

### PR DESCRIPTION
**Pull request by scottxu0730**
[WIP]This PR adds cost_aware_partition method in Partitioner class. The method partitions the fx graph module based on the latency of the whole graph.

### CODE REVIEW

File: torch/fx/experimental/Partitioner.py [hunk 1]
```diff
@@ -43,96 +46,14 @@ def create_node(
         node = DAGNode(submodule_node, input_nodes, output_nodes, logical_devices, size_bytes)
         self.nodes.append(node)
 
-class Partition:
```
Comment (scottxu0730) [inline torch/fx/experimental/Partitioner.py:46]: Move Partition class to partitioner_utils.py

File: torch/fx/experimental/Partitioner.py [hunk 2]
```diff
@@ -43,96 +46,14 @@ def create_node(
...
-                if all([n not in self.nodes for n in input_node.users]):
-                    self.nodes.remove(input_node)
-
 class PartitionResult(NamedTuple):
     """NameTuple used for returning DAG and a new graph module
     """
     dag: DAG
     module_with_submodules: GraphModule
 
-class Device(NamedTuple):
```
Comment (scottxu0730) [inline torch/fx/experimental/Partitioner.py:98]: Move Device class to partitioner_utils.py

File: torch/fx/experimental/Partitioner.py [hunk 3]
```diff
@@ -43,96 +46,14 @@ def create_node(
...
-    available_mem_bytes: int
-    logical_id: int
-
-class PartitionerConfig(NamedTuple):
-    devices: List[Device]
-    is_sparse_nn: bool = False
-
 """Followings are some helper functions for partition manipulation"""
 
-def get_extra_size_of(node: Node, nodes: Set[Node]) -> int:
```
Comment (scottxu0730) [inline torch/fx/experimental/Partitioner.py:109]: Move get_extra_size_of to partitioner_utils.py

**Review by gcatron — APPROVED** [review]
Nice work!

**Pull request closed without merging**